src/trails.erl

```diff
@@ -151,18 +153,61 @@ trails(Handler) ->
   trails([Handler], []).
 
 %% @doc Store the given list of trails.
--spec store(trails()) -> ok.
+-spec store(Trails::trails() |
+              [{HostMatch::route_match(), Trails::trails()}]) -> ok.
 store(Trails) ->
   _ = application:ensure_all_started(trails),
+  store('_', Trails).
+
+-spec store(Server::ranch:ref(),
+            Trails::trails() |
+              [{HostMatch::route_match(), Trails::trails()}]) -> ok.
+store(_Server, []) ->
+  ok;
+store(Server, [{HostMatch, Trails} | Hosts]) ->
+  NormalizedPaths = normalize_store_input(Trails),
+  do_store(Server, HostMatch, NormalizedPaths),
+  store(Server, Hosts);
+store(Server, Trails) when is_list(Trails) ->
   NormalizedPaths = normalize_store_input(Trails),
```

Comment:
This is the version of `store` that should do…
```erlang
_ = application:ensure_all_started(trails),
```
Make sure in one of your tests to call `trails:store/2` or `trails:store/3` with the application off, and it works anyway.